benchmarking/caching/simulation_cache.py

```diff
@@ -0,0 +1,115 @@
+import json
+import os
+from datetime import datetime
+from os import path
+from typing import List, Dict, Any
+import git
+from api.models.team_set import TeamSet
+
+
+class SimulationCache:
+    """
+    This class is used to cache the results of a simulation, which is a List[TeamSet], where each element of the list is a different run.
+    All TeamSets should have been generated by the same algorithm (with the same settings).
+    It uses the cache_key to generate a file with the results of the simulation.
+    """
+
+    def __init__(self, cache_key: str) -> None:
+        self.cache_key = cache_key
+        self._data = None
+        """
+        Dict in the form of:
+        {
+            "metadata": Dict[str, Any],  # Contains at least "timestamp" and "commit_hash"
+            "team_sets": List[TeamSet],
+        }
+        """
+
+    def exists(self) -> bool:
+        """
+        Checks if the cache_key is in the cache.
+        """
+        return path.exists(self._get_file())
+
+    def get_teams(self) -> List[TeamSet]:
+        """
+        Gets the simulation results from the cache.
+        """
+        self._load_data()
+
+        return self._data["team_sets"]
+
+    def get_metadata(self) -> Dict[str, Any]:
+        """
+        Gets the metadata associated with the simulation results from the cache.
+        """
+        self._load_data()
+
+        return self._data["metadata"]
+
+    def save(
+        self, simulation_results: List[TeamSet], metadata: Dict[str, Any] = None
+    ) -> None:
+        """
+        Puts the simulation results into the cache.
+        Overwrites any existing cache with the same cache_key.
+        """
+
+        # Get metadata
+        metadata = metadata or {}
+        metadata["timestamp"] = datetime.now().isoformat()
+        # Get latest commit hash. This is so that we can track which commit generated the cache and go back to the code that generated it if needed. Would then be accessible at https://github.com/Teamable-Analytics/algorithms/commit/<commit_hash>
+        metadata["commit_hash"] = git.Repo(
+            search_parent_directories=True
+        ).head.object.hexsha
```

Comment:
These hashes are actually independent of the branch they're on, so this works on both main and any other branch